plugins/woocommerce/tests/php/src/Blocks/Domain/Services/CheckoutFieldsFrontendTest.php

```diff
@@ -0,0 +1,263 @@
+ */
+class CheckoutFieldsFrontendTest extends TestCase {
+	/**
+	 * System under test.
+	 *
+	 * @var CheckoutFieldsFrontend
+	 */
+	private $sut;
+
+	/**
+	 * Helper controller
+	 *
+	 * @var CheckoutFields
+	 */
+	private $controller;
+
+	/**
+	 * Setup.
+	 */
+	public function setUp(): void {
+		parent::setUp();
+
+		add_filter( 'woocommerce_add_notice', [ $this, 'capture_notice' ] );
+		add_filter( 'woocommerce_add_error', [ $this, 'capture_error' ] );
+		add_filter( 'woocommerce_add_success', [ $this, 'capture_success' ] );
+
+		$this->sut        = Package::container()->get( CheckoutFieldsFrontend::class );
+		$this->controller = Package::container()->get( CheckoutFields::class );
+	}
+
+	/**
+	 * Tear down.
+	 */
+	protected function tearDown(): void {
+		remove_filter( 'woocommerce_add_notice', [ $this, 'capture_notice' ] );
+		remove_filter( 'woocommerce_add_error', [ $this, 'capture_error' ] );
+		remove_filter( 'woocommerce_add_success', [ $this, 'capture_success' ] );
+	}
+
+	/**
+	 * Capture success messages.
+	 *
+	 * @param string $message The message string.
+	 */
+	public function capture_success( string $message ): void {
+		$this->capture_messages( $message, 'success' );
+	}
+
+	/**
+	 * Capture error messages.
+	 *
+	 * @param string $message The message string.
+	 */
+	public function capture_error( string $message ): void {
+		$this->capture_messages( $message, 'error' );
+	}
+
+	/**
+	 * Capture notice messages.
+	 *
+	 * @param string $message The message string.
+	 */
+	public function capture_notice( string $message ): void {
+		$this->capture_messages( $message, 'notice' );
+	}
+
+	/**
+	 * Capture messages.
+	 *
+	 * @param string $message The message string.
+	 * @param string $type The message type.
+	 */
+	private function capture_messages( string $message, string $type ): void {
+		global $mocked_messages;
+		$mocked_messages[] = [
+			'message' => $message,
+			'type'    => $type,
+		];
+	}
+
+
+	/**
+	 * @testDox Save account form fields.
+	 */
+	public function test_save_account_form_fields_contact_customer_data_store_exception() {
+
+		$hash = uniqid();
+
+		$mock_data_store = function () use ( $hash ) {
+			throw new Exception( $hash ); // phpcs:ignore WordPress.Security.EscapeOutput.ExceptionNotEscaped
+		};
+
+		add_filter( 'woocommerce_customer_data_store', $mock_data_store );
+
+		global $mocked_messages;
+		$mocked_messages = [];
+
+		$this->sut->save_account_form_fields( 12345 );
+
+		remove_filter( 'woocommerce_customer_data_store', $mock_data_store );
+
+		$this->assertCount( 1, $mocked_messages );
+		$this->assertEquals( __( 'Unable to save customer additional fields: ', 'woocommerce' ) . $hash, $mocked_messages[0]['message'] );
+		$this->assertEquals( 'notice', $mocked_messages[0]['type'] );
+	}
+
+	/**
+	 * @testDox Contact additional field validation error
+	 */
+	public function test_save_account_form_fields_contact_generic_validation_error() {
```

Comment:
I think we should add another test where we check that an optional field with validation, submitted with an empty value can be saved correctly.

Reply:
Added 👍🏼